File: Py Scripts/GoalScript.py
import UnityEngine
import UnityEngine.Events

class GoalScript(UnityEngine.MonoBehaviour):
    ballTag = "Ball"
    onTriggerEnter = UnityEngine.Events.UnityEvent()

    def OnTriggerEnter(self, other):
        if other.CompareTag(self.ballTag):
            print("GOOAAALLLL")
            other.gameObject.SetActive(False)
            self.onTriggerEnter.Invoke()

File: scripts/GoalScript.cs
using UnityEngine;
using UnityEngine.Events;

public class GoalScript : MonoBehaviour
{
   
    public string ballTag = "Ball";
    
    public UnityEvent onTriggerEnter;
  


  
    private void OnTriggerEnter(Collider other)
    {
        if (other.CompareTag(ballTag))
        {
            Debug.Log("GOOAAALLLL");
            other.gameObject.SetActive(false);
            onTriggerEnter.Invoke();
            
        }
    }
}